ci-operator/config/openshift/cincinnati/openshift-cincinnati-master.yaml

```diff
@@ -3,58 +3,66 @@ base_images:
+    tag: boot-root-1.79.0
 images:
 - dockerfile_literal: |
-    FROM ubi
-    COPY . /opt/app-root/src/cincinnati/
-    WORKDIR /opt/app-root/src/cincinnati
-    USER 0
-    ENV HOME="/root"
-    ENV PATH="${HOME}/.cargo/bin:${HOME}/.local/bin:${PATH}"
-    RUN chgrp root /opt/app-root/src/cincinnati && \
-        chmod g+w /opt/app-root/src/cincinnati && \
-        dnf install -y jq gcc openssl-devel && \
-        dnf clean all
-    # build: Rust stable toolchain
-    RUN curl https://sh.rustup.rs -sSf | sh -s -- --default-toolchain 1.70.0 -y && \
-      rustup install 1.64.0
-    # install yamllint
-    RUN python3 -m ensurepip --upgrade && pip3 install --user yamllint
-    RUN \
-      mkdir -p $HOME/.cargo/git/ && \
-      find $HOME/. -type d -exec chmod 777 {} \; && \
-      find $HOME/. -type f -exec chmod ugo+rw {} \;
-    RUN curl -LsSf https://get.nexte.st/latest/linux | tar zxf - -C ${CARGO_HOME:-~/.cargo}/bin
-    USER 1001
+    FROM registry.access.redhat.com/ubi9/ubi:latest
+    WORKDIR /root
+    ENV HOME=/root PATH="/root/.cargo/bin:${PATH}"
+    RUN dnf update -y \
+        && dnf install -y git gcc openssl-devel \
+        && dnf clean all \
+        && curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+    COPY . /root/cincinnati/
+    WORKDIR /root/cincinnati
+    RUN cargo build --release
+  from: ubi
+  to: rustup-build
+- dockerfile_literal: |
+    FROM registry.access.redhat.com/ubi9/ubi:latest
+    ENV RUST_LOG=actix_web=error,dkregistry=error
+    COPY --from=pipeline:bin /go/bin/graph-builder /go/bin/policy-engine /go/bin/metadata-helper /usr/bin/
+    ENTRYPOINT ["/usr/bin/graph-builder"]
   from: ubi
-  to: src-rust
-- dockerfile_path: dist/Dockerfile.rust-toolset/Dockerfile
   inputs:
-    ubi:
+    bin:
       as:
-      - registry.access.redhat.com/ubi9/ubi:latest
+      - pipeline:bin
   to: deploy
-- dockerfile_path: dist/Dockerfile.e2e-ubi/Dockerfile
+- dockerfile_literal: |
+    FROM registry.ci.openshift.org/cincinnati-ci-public/rust:e2e-test-ubi9
+    ENV HOME="/root"
+    WORKDIR "${HOME}/cincinnati"
+    COPY --from=pipeline:test-bin /opt/cincinnati/bin/e2e /usr/bin/cincinnati-e2e-test
+    COPY --from=pipeline:test-bin /opt/cincinnati/bin/prometheus_query /usr/bin/cincinnati-prometheus_query-test
+    COPY --from=pipeline:test-bin /opt/cincinnati/bin/slo /usr/bin/cincinnati-e2e-slo
+    COPY hack/e2e.sh hack/vegeta.targets hack/
+    COPY dist/openshift/cincinnati-deployment.yaml \
+         dist/openshift/cincinnati-e2e.yaml \
+         dist/openshift/observability.yaml \
+         dist/openshift/load-testing.yaml \
+         dist/openshift/
+    COPY e2e/tests/testdata e2e/tests/testdata
+    COPY dist/prepare_ci_credentials.sh dist/cargo_test.sh dist/
+    ENV E2E_TESTDATA_DIR "e2e/tests/testdata"
+    ENTRYPOINT ["hack/e2e.sh"]
+  from: e2e
   inputs:
-    e2e:
+    test-bin:
       as:
-      - registry.ci.openshift.org/cincinnati-ci-public/rust:e2e-test-ubi9
-    rust:
```

Comment:
`rust`: it was a mistake by me in https://github.com/openshift/release/commit/ddbe54d49cbfa0a9e743c73543e3dcc6904d7d5c
to use it as `images[].inputs`. It should be `images[].from` instead. It leads to the result that `rust` is no-op in the build and the build in CI pull the image in the `FROM` directive, instead of using a tag in CI registry.